packages/opentelemetry-core/src/common/ConsoleLogger.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Logger } from '@opentelemetry/types';
+import * as consoleLogLevel from 'console-log-level';
+import * as util from 'util';
+
+/** Defines a logger options interface. */
+export interface LoggerOptions {
+  /** level of logger - 0:disable, 1: error, 2: warn, 3: info, 4: debug  */
+  level?: string;
+}
+export type ConsoleLogLevel = 'error' | 'warn' | 'info' | 'debug';
+export type LogLevel = 'silent' | ConsoleLogLevel;
+
+/** The list of log levels. */
+export const LEVELS: ReadonlyArray<LogLevel> = [
+  'silent',
+  'error',
+  'warn',
+  'info',
+  'debug',
+];
+
+export class ConsoleLogger implements Logger {
+  private _logger: consoleLogLevel.Logger;
+  level?: string;
+  constructor(options?: LoggerOptions | string | number) {
+    this.level = this._logLevelToName(options).level || 'debug';
+    this._logger = consoleLogLevel({
+      stderr: true,
+      level: this.level as ConsoleLogLevel,
+    });
+  }
+
+  debug(message: string, ...args: unknown[]) {
+    this._logger.debug(util.format(message, ...args));
+  }
+
+  error(message: string, ...args: unknown[]) {
+    this._logger.error(util.format(message, ...args));
+  }
+
+  warn(message: string, ...args: unknown[]) {
+    this._logger.warn(util.format(message, ...args));
+  }
+
+  info(message: string, ...args: unknown[]) {
+    this._logger.info(util.format(message, ...args));
+  }
+
+  private _logLevelToName(
+    options?: LoggerOptions | string | number
+  ): LoggerOptions {
+    if (typeof options === 'number') {
+      if (options < 0) options = 0;
+      else if (options >= LEVELS.length) options = LEVELS.length - 1;
+      return { level: LEVELS[options] };
+    } else if (typeof options === 'string') {
+      return { level: options };
+    } else {
+      return options || {};
+    }
+  }
+}
+
+/**
+ * Function logger exported to others classes. Inspired by:
```

Comment:
While I can see how `logger('silent')` is *slightly* shorter than `new ConsoleLogger('silent')`, I think I would prefer just exporting the class and not having this intermediate creation function.

How do you feel about it?